src/main/cpp/src/genomicsdb/variant_field_handler.cc

```diff
@@ -146,10 +189,215 @@ void  VariantOperations::remap_data_based_on_genotype(const std::vector<DataType
   }
 }
 
+
+/*
+ * Reorders fields whose length and order depend on the number of genotypes (BCF_VL_G)
+ * for general ploidy
+ */
+template<class DataType>
+void VariantOperations::remap_data_based_on_genotype_general(const std::vector<DataType>& input_data,
+    const uint64_t input_call_idx,
+    const CombineAllelesLUT& alleles_LUT,
+    const unsigned num_merged_alleles, bool NON_REF_exists, const unsigned ploidy,
+    RemappedDataWrapperBase& remapped_data,
+    std::vector<uint64_t>& num_calls_with_valid_data, DataType missing_value,
+    std::vector<int>& remapped_allele_idx_vec_for_current_gt_combination,
+    std::vector<std::pair<int, int> >& ploidy_index_allele_index_stack,
+    std::vector<int>& input_call_allele_idx_vec_for_current_gt_combination,
+    remap_operator_function_type<DataType> op
+)
+{
+  //index of NON_REF in merged variant
+  const auto merged_non_reference_allele_idx = NON_REF_exists ? 
+    static_cast<int64_t>(static_cast<int>(num_merged_alleles-1)) : lut_missing_value;
+  //index of NON_REF in input sample
+  const auto input_non_reference_allele_idx = NON_REF_exists ? 
+    alleles_LUT.get_input_idx_for_merged(input_call_idx, merged_non_reference_allele_idx) : lut_missing_value;
+#define SET_PLOIDY_INDEX_IN_STACK_ELEMENT(X,Y) ((X).first = (Y))
+#define SET_ALLELE_INDEX_IN_STACK_ELEMENT(X,Y) ((X).second = (Y))
+#define GET_PLOIDY_INDEX_IN_STACK_ELEMENT(X) ((X).first)
+#define GET_ALLELE_INDEX_IN_STACK_ELEMENT(X) ((X).second)
+  remapped_allele_idx_vec_for_current_gt_combination.resize(ploidy+1u); //+1 to avoid unnecessary if statements in the while loop
+  input_call_allele_idx_vec_for_current_gt_combination.resize(ploidy);
+  //Enumerate genotypes based on method described in 
+  //http://genome.sph.umich.edu/wiki/Relationship_between_Ploidy,_Alleles_and_Genotypes
+  //Use custom "stack" instead of a recursive function call
+  //"top" of the stack is the last element of the vector
+  assert(ploidy > 0u);
```

Comment:
move this to the beginning of the function. Why proceed if this check fail?

Reply:
Fixed